From bafb120eba6d958d448427de53661d9c13506f93 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 17 Sep 2015 15:44:20 -0500
Subject: [PATCH 23/97] apply backport patch
 collateral-evolutions/network/0003-cfg80211-wext-padding/net_wireless_core.patch

---
 backports/net/wireless/core.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/backports/net/wireless/core.c b/backports/net/wireless/core.c
index 2a0bbd2..0662198 100644
--- a/backports/net/wireless/core.c
+++ b/backports/net/wireless/core.c
@@ -344,6 +344,17 @@ struct wiphy *wiphy_new_nm(const struct cfg80211_ops *ops, int sizeof_priv,
 	struct cfg80211_registered_device *rdev;
 	int alloc_size;
 
+	/*
+	 * Make sure the padding is >= the rest of the struct so that we
+	 * always keep it large enough to pad out the entire original
+	 * kernel's struct. We really only need to make sure it's larger
+	 * than the kernel compat is compiled against, but since it'll
+	 * only increase in size make sure it's larger than the current
+	 * version of it. Subtract since it's included.
+	 */
+	BUILD_BUG_ON(WIPHY_COMPAT_PAD_SIZE <
+		     sizeof(struct wiphy) - WIPHY_COMPAT_PAD_SIZE);
+
 	WARN_ON(ops->add_key && (!ops->del_key || !ops->set_default_key));
 	WARN_ON(ops->auth && (!ops->assoc || !ops->deauth || !ops->disassoc));
 	WARN_ON(ops->connect && !ops->disconnect);
-- 
2.5.1

